z2/src/converter.rs

```diff
@@ -248,8 +237,8 @@ pub async fn convert_persistence(
                         });
 
                         let receipt = TransactionReceipt {
-                            block_hash: block.hash(),
                             tx_hash: Hash(txn_hash.0),
+                            block_hash: Hash::ZERO,
```

Comment:
This is what I meant about circular hashing and the fact that we should have a comment to say in which order we are doing it at the start of my review comments, FWIW :-)

Reply:
I think it will be cleaner in future to add a constructor to `TransactionReceipt` which takes the non-block hash fields, calculates the block hash then populates the struct. I think this kind of pre-initialisation is a bit of a smell. But I can accept if we do that later since this PR has been around for a while.